### PR TITLE
LPS-93141 Add tests for data definition rule functions

### PR DESCRIPTION
**Pull request by mclomello**

### CODE REVIEW

**Comment by mclomello** [comment]
ci:test:sf

**Comment by mclomello** [comment]
ci:test:relevant

**Comment by jeyvison** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/data-engine/data-engine-rest-impl/src/test/java/com/liferay/data/engine/rest/internal/rule/function/v1_0/DataDefinitionRulesEmailAddressResourceTest.java [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+
+/**
+ * @author Marcelo Mello
+ */
+public class DataDefinitionRulesEmailAddressResourceTest
+	extends DataDefinitionRulesTestCase {
+
+	@Before
+	public void setUp() {
+		_dataDefinition = randomDataDefinition();
```
Comment (jeyvison) [inline modules/apps/data-engine/data-engine-rest-impl/src/test/java/com/liferay/data/engine/rest/internal/rule/function/v1_0/DataDefinitionRulesEmailAddressResourceTest.java]: It doens't make sense to have a random definition that only vreates a new instance of data definition. Instantiate it here or add more code to randomDataDefinition() if it makes sense

File: modules/apps/data-engine/data-engine-rest-impl/src/test/java/com/liferay/data/engine/rest/internal/rule/function/v1_0/DataDefinitionRulesEmptyResourceTest.java [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+
+		_dataDefinition.setDataDefinitionRules(
+			this::_randomDataDefinitionRules);
+
+		_dataRecord = randomDataRecord();
+		_dataRuleFunctionTracker = name -> new EmptyDataRuleFunction();
+	}
+
+	@Test
+	public void testEmpty() {
```
Comment (jeyvison) [inline modules/apps/data-engine/data-engine-rest-impl/src/test/java/com/liferay/data/engine/rest/internal/rule/function/v1_0/DataDefinitionRulesEmptyResourceTest.java]: Why there is a testEmpty and testWithEmptyValue methods?

File: modules/apps/data-engine/data-engine-rest-impl/build.gradle [hunk 3]
```diff
@@ -29,4 +29,9 @@ dependencies {
 	compileOnly project(":apps:portal-vulcan:portal-vulcan-api")
 	compileOnly project(":core:petra:petra-function")
 	compileOnly project(":core:petra:petra-string")
+
+	testCompile group: "org.skyscreamer", name: "jsonassert", version: "1.2.3"
+	testCompile group: "org.slf4j", name: "slf4j-api", version: "1.7.22"
```
[the remark this answers is not in the document]
Comment (mclomello) [reply modules/apps/data-engine/data-engine-rest-impl/build.gradle]: Yes, at the bottom of testApply() method in deserialize class. But as we talk before that test will be removed and I'll remove this dependence to.

File: modules/apps/data-engine/data-engine-rest-impl/src/test/java/com/liferay/data/engine/rest/internal/rule/function/v1_0/DataDefinitionRulesDecimalLiteralResourceTest.java [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+ * @author Marcelo Mello
+ */
+public class DataDefinitionRulesDecimalLiteralResourceTest
+	extends DataDefinitionRulesTestCase {
+
+	@Before
+	public void setUp() {
+		_dataDefinition = new DataDefinition();
+
+		_dataDefinition.setDataDefinitionFields(_randomDataDefinitionFields());
```
[the remark this answers is not in the document]
Comment (mclomello) [reply modules/apps/data-engine/data-engine-rest-impl/src/test/java/com/liferay/data/engine/rest/internal/rule/function/v1_0/DataDefinitionRulesDecimalLiteralResourceTest.java]: I'll move the random method for base class receiving the field name and field type parameters.

File: modules/apps/data-engine/data-engine-rest-impl/src/test/java/com/liferay/data/engine/rest/internal/rule/function/v1_0/DataDefinitionRulesDecimalLiteralResourceTest.java [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+				}
+			});
+
+		Map<String, DataDefinitionField> dataDefinitionFields = Stream.of(
+			_dataDefinition.getDataDefinitionFields()
+		).collect(
+			Collectors.toMap(DataDefinitionField::getName, Function.identity())
+		);
+
+		DataDefinitionField dataDefinitionField;
```
[the remark this answers is not in the document]
Comment (mclomello) [reply modules/apps/data-engine/data-engine-rest-impl/src/test/java/com/liferay/data/engine/rest/internal/rule/function/v1_0/DataDefinitionRulesDecimalLiteralResourceTest.java]: Ok. I'll fix this.

**Pull request closed without merging**